openff/cli/generate_conformers.py

```diff
@@ -0,0 +1,346 @@
+
+import numpy as np
+from openforcefield.topology.molecule import Molecule, UndefinedStereochemistryError
+from openforcefield.typing.engines.smirnoff import ForceField
+from openforcefield.utils.toolkits import ToolkitRegistry
+from simtk import openmm, unit
+
+
+def generate_conformers(
+    molecule: str,
+    registry: ToolkitRegistry,
+    forcefield: str,
+    constrained: bool = False,
+    prefix: Optional[str] = None,
+) -> List[Molecule]:
+
+    ff_name = forcefield
+    if constrained:
+        split_name = ff_name.split("-")
+        split_name[0] = "openff_unconstrained"
+        ff_name = "-".join(split_name)
+    if not ff_name.endswith(".offxml"):
+        ff_name += ".offxml"
+    ff = ForceField(ff_name)
+
+    file_format = molecule.split(".")[-1]
+
+    # TODO: This may not preserve order of loading molecules in
+    ambiguous_stereochemistry = False
+    try:
+        raw_mols = registry.call(
+            "from_file",
+            molecule,
+            file_format=file_format,
+        )
+    except UndefinedStereochemistryError:
+        ambiguous_stereochemistry = True
+        raw_mols = registry.call(
+            "from_file",
+            molecule,
+            file_format=file_format,
+            allow_undefined_stereo=True,
+        )
+
+    # When failing to parse molecules (i.e. attempting to read MOL2 with
+    # RDKit, which is not supported) the toolkit can return an empty
+    # list instead of raising a specific exception
+    if raw_mols == []:
+        from openff.cli.utils.exceptions import MoleculeParsingError
+
+        raise MoleculeParsingError(toolkit_registry=registry, filename=molecule)
+
+    mols = []
+    for i, mol in enumerate(raw_mols):
+        if prefix is not None:
+            mol.name = prefix
+        elif not mol.name:
+            mol.name = "molecule"
+        if len(raw_mols) > 1:
+            mol.name += str(i)
+        mols.append(mol)
+
+    # TODO: How to handle names of different stereoisomers? Just act like they're different conformers?
+    if ambiguous_stereochemistry:
+        mols_with_unpacked_stereoisomers = []
+        for mol in mols:
+            # TODO: This is a brute-force approach, it would be better to check stereo
+            #  without needing to call enumerate_stereoisomers
+            stereoisomers = mol.enumerate_stereoisomers()
+            if stereoisomers:
+                for i, iso in enumerate(stereoisomers):
+                    iso.name = mol.name + "_stereoisomer" + str(i)
+                    mols_with_unpacked_stereoisomers.append(iso)
+            else:
+                mols_with_unpacked_stereoisomers.append(mol)
+        mols = mols_with_unpacked_stereoisomers
+
+    for mol in mols:
+        # TODO: Keep some conformers, if found in input molecule?
+        mol.generate_conformers(
+            toolkit_registry=registry,
+            n_conformers=20,
+            rms_cutoff=0.25 * unit.angstrom,
+        )
```

Comment:
(not blocking) I think so. However, generate_conformers will delete existing conformers. To keep them, we could use something like 
```suggestion
    for mol in mols:
        existing_conf = None
        if mol.conformers:
           existing_conf = copy.deepcopy(mol.conformers[0])
        mol.generate_conformers(
            toolkit_registry=registry,
            n_conformers=100,
            rms_cutoff=0.25 * unit.angstrom,
        )
        if existing_conf:
            mol.add_conformers(existing_conf)
```